src/careamics/models/lvae/lvae.py

```diff
@@ -17,42 +18,80 @@
 from .layers import (
     BottomUpDeterministicResBlock,
     BottomUpLayer,
+    GateLayer,
     TopDownDeterministicResBlock,
     TopDownLayer,
 )
-from .utils import Interpolate, ModelType, crop_img_tensor, pad_img_tensor
+from .utils import Interpolate, ModelType, crop_img_tensor
 
 
 @register_model("LVAE")
 class LadderVAE(nn.Module):
+    """
+    Constructor.
+
+    Parameters
+    ----------
+    input_shape : int
+        The size of the input image.
+    output_channels : int
+        The number of output channels.
+    multiscale_count : int
+        The number of scales for multiscale processing.
+    z_dims : list[int]
+        The dimensions of the latent space for each layer.
+    encoder_n_filters : int
+        The number of filters in the encoder.
+    decoder_n_filters : int
+        The number of filters in the decoder.
+    encoder_conv_strides : list[int]
+        The strides for the conv layers encoder.
+    decoder_conv_strides : list[int]
+        The strides for the conv layers decoder.
+    encoder_dropout : float
+        The dropout rate for the encoder.
+    decoder_dropout : float
+        The dropout rate for the decoder.
+    nonlinearity : str
+        The nonlinearity function to use.
+    predict_logvar : bool
+        Whether to predict the log variance.
+    analytical_kl : bool
+        Whether to use analytical KL divergence.
+
+    Raises
+    ------
+    NotImplementedError
+        If only 2D convolutions are supported.
+    """
 
     def __init__(
         self,
         input_shape: int,
         output_channels: int,
         multiscale_count: int,
-        z_dims: List[int],
+        z_dims: list[int],
         encoder_n_filters: int,
         decoder_n_filters: int,
+        encoder_conv_strides: list[int],
+        decoder_conv_strides: list[int],
         encoder_dropout: float,
         decoder_dropout: float,
         nonlinearity: str,
         predict_logvar: bool,
         analytical_kl: bool,
     ):
-        """
-        Constructor.
-
-        Parameters
-        ----------
-
-        """
         super().__init__()
 
         # -------------------------------------------------------
         # Customizable attributes
         self.image_size = input_shape
+        """Input image size. (C, Z, Y, X) or (C, Y, X) if the data is 2D."""
```

Comment:
Need clarification here: Channel dimension is now part of `image_size`? If yes, what do we expect it to be? In my understanding, it should be `multiscale_count`. 

But if this is the case then we would have redundant info in this attribute!

Reply:
Multiscale_count is different. But channel shouldn't be there